xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -117,6 +117,7 @@ SelectionStream& CSelectionStreams::Get(StreamType type, int index)
 
 std::vector<SelectionStream> CSelectionStreams::Get(StreamType type)
 {
+  CSingleLock lock(m_section);
```

Comment:
So instead of the race condition fix, will you accept a patch that adds API documentation stating that using this method makes an instances thread-unsafe?